components/nexus-core/src/main/java/org/sonatype/nexus/proxy/item/ChecksumReconciler.java

```diff
@@ -114,7 +129,46 @@ void reconcileItemChecksum(final Repository repo, final StorageFileItem item) th
         finally {
           uid.getLock().unlock();
         }
+
+        final String itemPath = item.getPath();
+        if (!itemPath.endsWith(".sha1") && !itemPath.endsWith(".md5")) {
```

Comment:
And checksum files themselves also have their own checksums stored in item attributes, which also need cleaning up - it's just that those are typically not surfaced in the local storage system.